src/path/dec_int.rs

```diff
@@ -55,6 +55,17 @@ impl DecInt {
     }
 }
 
+impl core::fmt::Write for DecInt {
+    fn write_str(&mut self, s: &str) -> core::fmt::Result {
+        if self.len + s.len() > 20 {
+            return Err(core::fmt::Error);
+        }
+        self.buf[self.len..self.len + s.len()].copy_from_slice(s.as_bytes());
```

Comment:
Instead of manual bounds-checking, would it work to call [`self.buf.get[self.len..self.len+s.len()]`](https://doc.rust-lang.org/stable/std/primitive.slice.html#method.get), and do a `match` on it to handle the error?